### DISCUSSION TITLE
Scorporo IVA per elenco pagamenti

### DISCUSSION THREAD

**Discussion by mauriziofonte** [fattureincloud/api#123] [Q&A] [answered]
Buonasera,

Spero di non fare uno strafalcione pazzesco, ma credo di aver riscontrato un errore strano e vorrei sapere la vostra in merito.

Dopo svariate analisi e tentativi, l'API risponde in maniera contrastante a volte con il messaggio di errore `Il totale dei pagamenti non corrisponde al totale da pagare`, altre volte senza.

Dopo un pò di test sono riuscito a capire che riguarda i campi `net_price` e `gross_price` del **Product**, controllati contro il `amount` del **Payment**.

Considerate questo esempio (`amount_vatted` arriva da Stripe come **integer**, da dividere per 100 per ottenere un numero a 2 cifre decimali). Considerate che è presente un solo `Product` nel payload inviato.

```
<?php
$amount_vatted = 14626;
$amount_devatted = round($amount_vatted / (1 + ($vat_percentage / 100)), 0); // 11989.0
$vat = ($amount_vatted - $amount_devatted); // 2637

$fic_net_price = round($amount_devatted/100, 2); // 119.89
$fic_gross_price = round($amount_vatted/100, 2); // 146.26
```

Con questi dati l'API di FIC risponderà con `Il totale dei pagamenti non corrisponde al totale da pagare`, perchè sto inviando `119.89 euro` come **netto** e `146.26 euro` come **lordo**.

La API come si comporta quando deve verificare che le somme dei netti corrispondono al `amount` del **Payment**?

In questo caso, `119.89 * 1.22 = 146.2658`. Credo che il check venga fatto su un round di `146.2658` che è effettivamente `146.27`

**Comment by valmoz** [comment, marked as answer]
Ciao @mauriziofonte hai già letto questa guida?
[Invoice Totals Calculation](https://developers.fattureincloud.it/docs/guides/invoice-totals)

**Reply by mauriziofonte** [reply]
Ciao @valmoz 

No, non l'avevo letta.
Quindi è sufficiente usare [https://developers.fattureincloud.it/docs/guides/invoice-totals/#-the-magic-field](https://developers.fattureincloud.it/docs/guides/invoice-totals/#-the-magic-field) ?

**Reply by emajo** [reply]
Ciao @mauriziofonte, prima di usare il _magic field_ che sistema automaticamente i pagamenti ti consiglio di provare a fare le chiamate per il calcolo dei totali descritte sempre su quella pagina per capire effettivamente dove è la differenza e se possibile risolvere senza il magic field per non andare in contro a comportamenti non voluti.

Grazie,
Emanuele.
